### PR TITLE
Use new HLL implementation in `DistinctStatistics`

### PR DESCRIPTION
**Pull request by lnkuiper**
After @hawkfish's excellent rework of our `approx_count_distinct` aggregate function in https://github.com/duckdb/duckdb/pull/12355, which uses HyperLogLog to estimate the unique value count, I've also replaced the HyperLogLog in `DistinctStatistics` in storage.

The new HLL is much smaller to store and hold in memory. DuckDB databases with the old HLL can still be read, but the old HLL will be discarded, and distinct counts (and, therefore, join ordering) will be wrong. Happy to receive feedback on this PR on options to improve this (if we want to).

### CODE REVIEW

**Comment by lnkuiper** [comment]
We will slightly regress on the Join Order Benchmark due to the changed HLL stats, but not much to do about it since we are losing precision in the new HLL. I've also managed to get a pretty accurate conversion from old to new and back!

File: src/include/duckdb/storage/statistics/distinct_statistics.hpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ class DistinctStatistics {
 public:
 	void Merge(const DistinctStatistics &other);
 
-	unique_ptr<DistinctStatistics> Copy() const;
+	unique_ptr<DistinctStatistics> Copy();
```
Comment (Mytherin) [inline src/include/duckdb/storage/statistics/distinct_statistics.hpp:35]: Instead of removing `const` here, we can make the lock `mutable`

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM - one minor comment:

**Comment by lnkuiper** [comment]
It has been difficult to get this PR to a state where we don't regress on join ordering or ingestion performance.

I've cranked up the sample rate on integral types so that our join ordering doesn't suffer, which has worked. However, the additional overhead of hashing those values and inserting them in the HLL has regressed ingestion performance noticeably.

I profiled data ingestion and found that in some cases, _half_ of the time is spent updating `NumericStats,` which is egregious. Why? We have 3 non-inlined function calls to update the statistics for every ingested value. I've restructured this a bit; now, all function calls are inlined. Ingestion is much faster, and I've increased the sample rate to 30% for integral values without regressions (I think!).

**Comment by lnkuiper** [comment]
Regression test failure is due to slightly different join orders on IMDB. We get better overall join orders on TPC-DS though so I don't think it's a problem

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**